### PR TITLE
Cooktop device

### PR DESCRIPTION
**Pull request by ap-whirlpool**
Add a new cooktop device to the matters devices xml file.

The new Cooktop Spec has the timer cluster removed. Here is the PR for the updated spec for this cooktop device:
https://github.com/CHIP-Specifications/connectedhomeip-spec/pull/8593

### CODE REVIEW

**Review by bzbarsky-apple — CHANGES_REQUESTED** [review]

File: src/app/zap-templates/zcl/data-model/chip/matter-devices.xml [hunk 1]
```diff
@@ -2257,4 +2257,18 @@ limitations under the License.
             </include>
         </clusters>
     </deviceType>
+    <deviceType>
+        <name>Cooktop</name>
+        <domain>CHIP</domain>
```
[the remark this answers is not in the document]
Comment (ap-whirlpool) [reply src/app/zap-templates/zcl/data-model/chip/matter-devices.xml:2262]: Thanks. Pushed.

File: src/app/zap-templates/zcl/data-model/chip/matter-devices.xml [hunk 2]
```diff
@@ -2257,4 +2257,18 @@ limitations under the License.
...
+        <typeName>Matter Cooktop</typeName>
+        <profileId editable="false">0x0103</profileId>
+        <deviceId editable="false">0x0078</deviceId>
+        <clusters lockOthers="true">
+            <include cluster="Identify" client="false" server="true" clientLocked="true" serverLocked="true"></include>
```
[the remark this answers is not in the document]
Comment (ap-whirlpool) [reply src/app/zap-templates/zcl/data-model/chip/matter-devices.xml:2267]: Should this be a client required? In the spec, it looks like both Identify and On/off are set to Server.

File: src/app/zap-templates/zcl/data-model/chip/matter-devices.xml [hunk 3]
```diff
@@ -2257,4 +2257,18 @@ limitations under the License.
...
+            <include cluster="On/Off" client="true" server="true" clientLocked="true" serverLocked="true">
```
[the remark this answers is not in the document]
Comment (ap-whirlpool) [reply src/app/zap-templates/zcl/data-model/chip/matter-devices.xml:2268]: Fixed this to be Server (not client).

File: src/app/zap-templates/zcl/data-model/chip/matter-devices.xml [hunk 4]
```diff
@@ -2257,4 +2257,18 @@ limitations under the License.
...
+                <requireAttribute>ON_OFF</requireAttribute>
+                <requireCommand>Off</requireCommand>
```
[the remark this answers is not in the document]
Comment (ap-whirlpool) [reply src/app/zap-templates/zcl/data-model/chip/matter-devices.xml:2270]: Yes, removed those 2 lines. Thanks!

File: src/app/zap-templates/zcl/data-model/chip/matter-devices.xml [hunk 5]
```diff
@@ -2257,4 +2257,18 @@ limitations under the License.
...
+            </include>
+        </clusters>
```
[the remark this answers is not in the document]
Comment (ap-whirlpool) [reply src/app/zap-templates/zcl/data-model/chip/matter-devices.xml:2271]: Good catch! The spec is being updated here:
https://github.com/CHIP-Specifications/connectedhomeip-spec/pull/8593

File: src/app/zap-templates/zcl/data-model/chip/matter-devices.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (eahove) [reply src/app/zap-templates/zcl/data-model/chip/matter-devices.xml:2268]: I believe this should be:
<include cluster="Identify" client="false" server="false" clientLocked="true" serverLocked="false"></include>
to show that the identify cluster is optional for the server.

File: src/app/zap-templates/zcl/data-model/chip/matter-devices.xml [hunk 6]
```diff
@@ -2257,4 +2257,17 @@ limitations under the License.
...
+    <deviceType>
+        <name>Cooktop</name>
+        <domain>CHIP</domain>
+        <typeName>Matter Cooktop</typeName>
+        <class>Simple</class>
+        <profileId editable="false">0x0103</profileId>
+        <deviceId editable="false">0x0078</deviceId>
+        <clusters lockOthers="true">
+            <include cluster="Identify" client="false" server="true" clientLocked="true" serverLocked="true"></include>
+            <include cluster="On/Off" client="false" server="true" clientLocked="true" serverLocked="true">
```
Comment (cecille) [inline src/app/zap-templates/zcl/data-model/chip/matter-devices.xml:2269]: The other device types also seem to include the descriptor cluster. Which seems fairly repetitive to me, tbh, but it seems like it might be needed since it's all over this file.

File: src/app/zap-templates/zcl/data-model/chip/matter-devices.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (cecille) [reply src/app/zap-templates/zcl/data-model/chip/matter-devices.xml:2268]: according to the zap folks, the 'server" and "client" attributes are how the attributes show up in the UI by default, and "locked" attributes indicate if they can be changed without a warning in the UI. For the most part, optionals are off by default, so 
```suggestion
            <include cluster="Identify" client="false" server="false" clientLocked="true" serverLocked="false"></include>
```

File: src/app/zap-templates/zcl/data-model/chip/matter-devices.xml (hunk 6 above)
Comment (ap-whirlpool) [reply src/app/zap-templates/zcl/data-model/chip/matter-devices.xml:2269]: Ok sounds good. Will put that in there

File: src/app/zap-templates/zcl/data-model/chip/matter-devices.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (ap-whirlpool) [reply src/app/zap-templates/zcl/data-model/chip/matter-devices.xml]: Done.

File: src/app/zap-templates/zcl/data-model/chip/matter-devices.xml [hunk 7]
```diff
@@ -2275,4 +2275,23 @@ limitations under the License.
...
+        <typeName>Matter Cooktop</typeName>
+        <class>Simple</class>
+        <profileId editable="false">0x0103</profileId>
+        <deviceId editable="false">0x0078</deviceId>
+        <clusters lockOthers="true">
+            <include cluster="Descriptor" client="false" server="true" clientLocked="true" serverLocked="true">
+                <requireAttribute>DEVICE_TYPE_LIST</requireAttribute>
+                <requireAttribute>SERVER_LIST</requireAttribute>
+                <requireAttribute>CLIENT_LIST</requireAttribute>
+                <requireAttribute>PARTS_LIST</requireAttribute>
```
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/data-model/chip/matter-devices.xml:2290]: Why are these here?  They don't need to be here; those are always mandatory attributes on Descriptor, no?

And yes, I know some other (non-spec, note) device types in this file do this.  Those are wrong.

File: src/app/zap-templates/zcl/data-model/chip/matter-devices.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (bzbarsky-apple) [reply src/app/zap-templates/zcl/data-model/chip/matter-devices.xml]: Right, the key part is this: optional things should never be locked.  `serverLocked="true"` means that the thing is either required or disallowed, depending on the value of `server`.

**Review by bzbarsky-apple — APPROVED** [review]

File: src/app/zap-templates/zcl/data-model/chip/matter-devices.xml (hunk 7 above)
Comment (ap-whirlpool) [reply src/app/zap-templates/zcl/data-model/chip/matter-devices.xml]: Thanks. Done. Pushed.

**Review by yunhanw-google — APPROVED** [review]
